src/python/WMCore/WMSpec/WMWorkload.py

```diff
@@ -956,7 +956,18 @@ def setProcessingVersion(self, processingVersions):
             task.setProcessingVersion(processingVersions, stepChainMap=stepNameMapping)
 
         self.updateLFNsAndDatasets()
-        self.data.properties.processingVersion = processingVersions
+        procVer = 0 # type declaration
+        if isinstance(processingVersions, str) or isinstance(processingVersions, bytes):
```

Comment:
ProcessingVersion is already validated at the WMSpec/StdBase module, there is no need for all these changes either.

Reply:
Yes, it could be anything **in** this method. But this setter is called from StdBase, which performs the input data sanitization/type already.
This is exactly what I mentioned a minute ago, which I would like to avoid checking data type every time that data flows between functions/methods. We should put a check at the very top level, and then all the bottom ones would be correct.

Of course, it does not give us protection for someone, one day, making a mistake and overwriting it along the code to a different data type... (or maybe it does at db level?)

Reply:
ok, done, and I added appropriate docstring about StdBae.